python/gherkin/gherkin_line.py

```diff
@@ -59,7 +59,7 @@ def split_table_cells(self, row):
                 cell = ''
                 start_col = col + 1
             elif char == '\\':
-                char = next(row)
+                char = next(row, "")
```

Comment:
An explicit `None` here would communicate that this is the end of the string and there are no more characters.

Reply:
I found the implementation a little unintuitive here, but recalling it gives the same outcome with either an `""` or `None` as both evaluate the same in the if statements i.e. if there's an escape character with nothing after it, it just appends the escape character only.

I kept with an empty string here, as the common operation is to work with the text and append the string value - an operation that would raise an error with `None`, though doesn't happen in this case as the conditionals reassign the value.

https://github.com/cucumber/gherkin/blob/f5cbb5d7f52de872eb881526ea5eff4a980bbd7e/python/gherkin/gherkin_line.py#L61-L69

However, what may be more explicit, is to use the `None` you mentioned and to `break`.

```suggestion
                char = next(row, None)
                if not char:
                    break
```

So suggestion would be to leave as empty string if staying with existing implementation, or to introduce conditional to break with explicit `None`. I'll go with your preference on that.

Reply:
What about not reusing `char` in this block? And instead calling it escaped?

```python
            elif char == '\\':
                escaped = next(row, "")                                
                col += 1                            
                if escaped == 'n':
                    cell += '\n'
                else:
                    if escaped not in ['|', '\\']:
                        cell += '\\'
                    cell += escaped
```

It would make it easier to see that we escape the end-of-line.

Reply:
Nevermind. I had a look at the JS and Ruby implementations. This is fine as is. It's the Java implementation that is the odd one out.